### ISSUE TITLE
Custom gift response template

### ISSUE THREAD

**Issue by pcsbx** [StardewModders/mod-ideas#55]
I'd like to write a small mod to refine gift responses.

I can't code though so I'd like a SMAPI mod template that I could modify.

It should:
- override vanilla responses for given item IDs and item group IDs
- override vanilla tastes for given item IDs and item group IDs
- not mess up vanilla tastes/responses for any items/groups that are not overriden by the mod.
- **bonus**: option to define a completely customized reaction (see below).

Example:
```
Abigail {-2} like "Wow, thanks! How did you know I was clamouring to lick a pretty stone?"
Clint {80} hate "Oh... the jolly wee berry of Winter Star's merry..."
```
Fully-custom reaction format:
```
<name> <item/group ID> <Δ friendship> <reaction portrait> <reaction text>
```

**Comment by Pathoschild** [comment]
Custom gift tastes are pretty straightforward in SMAPI 2.0+. Custom responses are harder, but not prohibitively so; the mod would probably detect the gift-response dialogue, figure out which item was given, and change the dialogue text.

Your proposed reaction format doesn't take translations into account, though (unless you redefine the whole line for each language).

**Comment by pcsbx** [comment]
Wouldn't it be best for possible translated versions to just be versions of the original mod file with translated response text, so someone playing in German would use the German version of the mod (if available) and so on?

**Comment by Pathoschild** [comment]
Yes, that's what I meant by redefining the whole line for each language.

**Comment by CoxinhaX** [comment]
I think this is the mod https://www.nexusmods.com/stardewvalley/mods/7304

**Comment by tiakall** [comment]
1.6 now allows custom reactions to any gifted item (including the option to reject it.
